### DISCUSSION TITLE
Using Pico W?

### DISCUSSION THREAD

**Discussion by phil-barrett** [grblHAL/RP2040#48] [General]
I am checking out support for a Pico W but I'm not sure if I'm doing it right. I set up a build for Pico W, plugged in the ssid and password (blacked out in the screen cap).
![image](https://user-images.githubusercontent.com/16570958/216794884-6d631a7d-a809-4db2-90e3-434b9291ccd5.png)
I am not seeing the board on my wifi network after saving and power cycling. (I have a wifi app that shows all devices connected to my access point.) It should show up as grblHAL. right? I used micropython on the pico w to start up wifi and it showed up so the hardware is ok. Also, [IP:] shows up in the console output - no IP address was assigned, I assume.

Here is the console output at boot up, if that helps:

> [VER:1.1f.20230129:]
> [OPT:VNMSL2,35,1024,3,0]
> [AXS:3:XYZ]
> [NEWOPT:ENUMS,RT+,ES,TC,SED,RTC,WIFI]
> [FIRMWARE:grblHAL]
> [NVS STORAGE:*FLASH]
> [DRIVER:RP2040]
> [DRIVER VERSION:230125]
> [DRIVER OPTIONS:SDK_1.4.0]
> [BOARD:PicoCNC]
> [AUX IO:5,8,0,0]
> [WIFI MAC:28:CD:C1:06:DD:2A]
> [IP:]
> [G54:0.000,0.000,0.000]
> [G55:0.000,0.000,0.000]
> [G56:0.000,0.000,0.000]
> [G57:0.000,0.000,0.000]
> [G58:0.000,0.000,0.000]
> [G59:0.000,0.000,0.000]
> [G59.1:0.000,0.000,0.000]
> [G59.2:0.000,0.000,0.000]
> [G59.3:0.000,0.000,0.000]
> [G28:0.000,0.000,0.000]
> [G30:0.000,0.000,0.000]
> [G92:0.000,0.000,0.000]
> [TLO:0.000,0.000,0.000]
> [PRB:0.000,0.000,0.000:0]
> 

I also built grblHAL with access point and the ssid doesn't show up after configuring and power cycling.

Used web builder fwiw.

**Comment by terjeio** [comment]
I have hardcoded the authentication method to [WPA2_AES_PSK](https://github.com/grblHAL/RP2040/blob/563c4908647a232bf430b00121e5d7bed3c0fdf4/wifi.c#L540), could it be that? I might have to add a setting for configuring the method if so.

FYI recently my router refused to accept connections from the Pico - and none of the callbacks were entered so I could not figure out why. Rebooting the router fixed it.

> I also built grblHAL with access point and the ssid doesn't show up after configuring and power cycling.

I have to disable that option in the Web Builder, IIRC AP mode is not (yet?) supported by the Pico framework.

**Reply by phil-barrett** [reply]
My router is set up for WPA/WPA2.  Probably would be a good idea to add a setting.  

The micropython code doesn't really call out an auth method though could be buried in the 2nd line of the python code.  So it's not clear but this code connects (with the ssid and password changed) and the connected device shows up in the list of devices connected to the router. I'll try to dig a little deeper.
```
import network
wlan = network.WLAN(network.STA_IF)
wlan.active(True)
wlan.connect("SSID","PASSWORD")
print(wlan.isconnected())
```
